app/AttributeTypes/SiUnitAttribute.php

```diff
@@ -2,338 +2,27 @@
-                        'convert' => fn($v) => $v * 0.0002365882372125699,
-                        'label' => 'cup_us',
-                    ],
-                    [
-                        'symbol' => 'gal',
-                        'convert' => fn($v) => $v * 0.003785412,
-                        'label' => 'gallon_us',
-                    ],
-                    [
-                        'symbol' => 'mi³',
-                        'convert' => fn($v) => $v * pow(1609.3440000009, 3),
-                        'label' => 'cubic_mile',
-                    ],
-                ],
-            ],
-            'mass' => [
-                'default' => 'kilogram',
-                'units' => [
-                    [
-                        'symbol' => 'g',
-                        'convert' => fn($v) => $v * pow(10, -3),
-                        'label' => 'gram',
-                    ],
-                    [
-                        'symbol' => 'kg',
-                        'convert' => fn($v) => $v,
-                        'label' => 'kilogram',
-                    ],
-                    [
-                        'symbol' => 't',
-                        'convert' => fn($v) => $v * pow(10, 3),
-                        'label' => 'ton',
-                    ],
-                    [
-                        'symbol' => 'oz',
-                        'convert' => fn($v) => $v * 0.028349523125,
-                        'label' => 'ounce',
-                    ],
-                    [
-                        'symbol' => 'lb',
-                        'convert' => fn($v) => $v * 0.45359237,
-                        'label' => 'pound',
-                    ],
-                ],
-            ],
-            'temperature' => [
-                'default' => 'degree_celsius',
-                'units' => [
-                    [
-                        'symbol' => '°C',
-                        'convert' => fn($v) => $v,
-                        'label' => 'degree_celsius',
-                    ],
-                    [
-                        'symbol' => '°K',
-                        'convert' => fn($v) => $v - 273,15,
-                        'label' => 'degree_kelvin',
-                    ],
-                    [
-                        'symbol' => '°F',
-                        'convert' => fn($v) => ($v - 32) / 1,8,
-                        'label' => 'degree_fahrenheit',
-                    ],
-                    [
-                        'symbol' => '°R',
-                        'convert' => fn($v) => $v * 1.25,
-                        'label' => 'degree_réaumur',
-                    ],
-                ],
-            ],
-            // 'substance' => [],
-            // 'luminous' => [],
-            // 'energy' => [],
-            // 'discspace' => [
-            //     // b, kb, mb, gb, tb, pb
-            //     // bi, kibi, mibi, gibi, tibi, pibi
-            // ],
-        ];
-    }
```

Comment:
We both didnt like the verbosity of the config array here, so I tried to move this to their respective classes.
I could not get it to work recreating the simple array structure so I sticked with the "UnitSystem" approach.